STM/vertx/README.adoc

```diff
@@ -0,0 +1,179 @@
+OVERVIEW
+--------
+
+Another example of STM and Vert.x integration. The quickstart is divided into 3 parts each introducing
+a different facet of STM and Vert.x integration.
+
+Use Case 1 shows a single JVM running multiple vertx instances sharing the same volatile STM object.
+
+Use Case 2 uses the same STM object but makes it persistent and shows how to share it across address
+spaces.
+
+These two examples illustrate the different ways in which an application can be scaled whilst
+maintaining data consistency:
+
+ - verticle scaling by using better hardware so that more threads can do be used to service the
+   workload;
+ - horizontal scaling by using more servers so that the workload can be distributed to multiple JVMs
+
+Use case 3 shows a single JVM hosting multiple STM objects and a Trip verticle which listens for
+trip, theatre and taxi booking requests. STM is used manage concurrent access to the various STM
+objects.
+
+Use case 4 is an optional final stress test to check that STM correctly manages concurrent access
+to STM objects.
+
+First build a fat jar that contains all the classes needed to run both demos in a single jar:
+
+  mvn clean package
+
+Use Case 1: Scaling by adding more threads
+==========================================
+
+Characteristics:
+
+  - RECOVERABLE and EXCLUSIVE
+  - creates multiple vertx instances each instance using a handle to the same STM object
+  
+What it's good for:
+
+  - vertical scaling where adding better h/w is an option in order to support more threads in one JVM
+
+Features:
+
+This example shows a Theatre booking service:
+   - an STM object is used to maintain all Theatre bookings
+     * the STM object is volatile
+   - multiple vertx instances each listening on the same HTTP endpoint
+   - each vertx instance shares the same STM object
+   - all vertx instances run in the same address space
+     * concurrency is managed by the STM runtime
+   - shows vertical scaling
+
+Usage:
+------
+
+  mvn exec:java -Pdemo1
```

Comment:
What happens if you type

   mvn clean package exec:java -Pdemo1

I am using what we use on narayana master Apache Maven 3.3.9

The quickstarts are on Apache Maven 3.2.5